Command/CloneSiteCommand.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\PageBundle\Command;
+
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\PageBundle\Model\PageInterface;
+use Sonata\PageBundle\Model\SiteInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Command for cloning complete sites including all pages and blocks.
+ *
+ * @author Armin Weihbold <armin.weihbold@gmail.com>
+ * @author Christian Gripp <mail@core23.de>
+ */
+final class CloneSiteCommand extends BaseCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function configure()
+    {
+        $this
+            ->setName('sonata:page:clone-site')
+            ->setDescription('Clone a complete site including all their pages')
+            ->addOption('sourceId', 's', InputOption::VALUE_REQUIRED, 'Source site id', null)
+            ->addOption('destId', 'd', InputOption::VALUE_REQUIRED, 'Destination site id', null)
+            ->addOption('prefix', 'p', InputOption::VALUE_REQUIRED, 'Title prefix', null)
+            ->addOption('onlyHybrid', 'oh', InputOption::VALUE_OPTIONAL, 'only clone hybrid pages', null);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        if (!$input->getOption('sourceId')) {
+            $this->listAllSites($output);
+
+            throw new \InvalidArgumentException('Please provide a "--sourceId=SITE_ID" option.');
+        }
+
+        if (!$input->getOption('destId')) {
+            $this->listAllSites($output);
+
+            throw new \InvalidArgumentException('Please provide a "--destId=SITE_ID" option.');
+        }
+
+        if (!$input->getOption('prefix')) {
+            throw new \InvalidArgumentException('Please provide a "--prefix=PREFIX" option.');
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('onlyHybrid')) {
```

Comment:
Why not use a default value for this option? You could get rid of the else part, right?